src/dist/replication/meta_server/greedy_load_balancer.cpp

```diff
@@ -945,5 +964,56 @@ void greedy_load_balancer::report(const dsn::replication::migration_list &list,
         _recent_balance_copy_secondary_count->add(counters[COPY_SEC_COUNT]);
     }
 }
+
+std::string
+greedy_load_balancer::set_balancer_ignored_app_id_list(const std::vector<std::string> &args)
+{
+    static const std::string invalid_arguments("invalid arguments");
+    std::stringstream oss;
+    if (args.empty()) {
+        dsn::zauto_read_lock l(_balancer_ignored_apps_lock);
+        oss << "ignored_app_id_list: ";
+        std::copy(_balancer_ignored_apps.begin(),
+                  _balancer_ignored_apps.end(),
+                  std::ostream_iterator<app_id>(oss, ","));
+        std::string app_ids = oss.str();
+        app_ids[app_ids.size() - 1] = '\0';
+        return app_ids;
+    }
+
+    if (args.size() != 1) {
+        return invalid_arguments;
+    }
+
+    dsn::zauto_write_lock l(_balancer_ignored_apps_lock);
+    if (args[0] == "clear") {
+        _balancer_ignored_apps.clear();
+        return "clear ok";
+    }
+
+    std::vector<std::string> app_ids;
+    dsn::utils::split_args(args[0].c_str(), app_ids, ',');
+    if (args.size() == 0) {
+        return invalid_arguments;
+    }
+
+    std::set<app_id> app_list;
+    for (const std::string &app_id_str : app_ids) {
+        app_id app;
+        if (!dsn::buf2int32(app_id_str, app)) {
+            return invalid_arguments;
+        }
+        app_list.insert(app);
+    }
+    _balancer_ignored_apps = std::move(app_list);
+    return "set ok";
```

Comment:
- if `args` is empty, get all ignored app id, otherwise, `args` length will always be equal to 1
- if `args` = clear, clear all ignored app id, otherwise, it should be like `app_id1, app_id2 ...`

I think your design is just like other remote command, like `replica.mem-release-max-reserved-percentage`, no args means get the value, has args means set the value. However, it will be strange that you can get the value from a setter function. Besides, you combine set and clear in this function, but your remote command explanation doesn't show this usage `lb.ignored_app_list [app_id1,app_id2..]`, I don't you how to clear ignored list unless reading source code.

In conclusion, I suggest you separate getter, setter into two function, add a new remote command called `lb.query_ignored_app`, you can return app_id and app_name to make it more clearer. For original setter function, no args mean clear ignore app id,  has args means set ignore app id. If you insist on combining three functions into one function, rename the function, add more comments and fulfill remote command usage explanation.